src/framework/core/js/DataBinding.js

```diff
@@ -507,18 +578,19 @@ https://github.com/fluid-project/infusion/raw/main/Infusion-LICENSE.txt
      * structure containing
      *     segs: {String[]} An array of model path segments to be dereferenced in the target component (will become `modelSegs` in the final return)
      *     context: {String} An IoC reference to the component holding the model
-     * @param {Boolean} implicitRelay - <code>true</code> if the reference was being resolved for an implicit model relay - that is,
-     * whether it occured within the `model` block itself. In this case, references to non-model material are not a failure and will simply be resolved
+     * @param {Boolean} permitNonModel - <code>true</code> If `true`, references to non-model material are not a failure and will simply be resolved
      * (by the caller) onto their targets (as constants). Otherwise, this function will issue a failure on discovering a reference to non-model material.
-     * @return {Object} - A structure holding:
-     *    that {Component} The component whose model is the target of the reference. This may end up being constructed as part of the act of resolving the reference
-     *    applier {Component} The changeApplier for the component <code>that</code>. This may end up being constructed as part of the act of resolving the reference
-     *    modelSegs {String[]} An array of path segments into the model of the component
+     * @return {ParsedModelReference} - A structure holding:
+     *    that {Component|Any} The component whose model is the target of the reference. This may end up being constructed as part of the act of resolving the reference.
+     * in the case of a reference to "local record" material such as {arguments} or {source}, `that` may exceptionally be a non-component.
+     *    applier {ChangeApplier|Undefined} The changeApplier for the component <code>that</code>. This may end up being constructed as part of the act of resolving the reference
```

Comment:
Are we still using the `<code></code>` syntax or backticks for code references in comments?

Reply:
I'm not sure, do you have a preference? It looks like there is a markdown plugin for JSDoc so I imagine we could use either, or both ...

Reply:
There's probably too much of both cases to clean things up in this PR anyways, so we may as well defer this to another time. I don't have a strong preference either way, but the backticks are easier to write.